hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/datanodes/datanodes.tsx

```diff
@@ -233,20 +295,48 @@ export class Datanodes extends React.Component<Record<string, object>, IDatanode
   };
 
   render() {
-    const {dataSource, loading, totalCount, lastUpdated} = this.state;
+    const {dataSource, loading, totalCount, lastUpdated, selectedColumns, columnOptions} = this.state;
     const paginationConfig: PaginationConfig = {
       showTotal: (total: number, range) => `${range[0]}-${range[1]} of ${total} datanodes`,
       showSizeChanger: true,
       onShowSizeChange: this.onShowSizeChange
     };
     return (
-      <div className='datanodes-container'>
-        <div className='page-header'>
+      <div className="datanodes-container">
+        <div className="page-header">
           Datanodes ({totalCount})
-          <AutoReloadPanel isLoading={loading} lastUpdated={lastUpdated} togglePolling={this.autoReload.handleAutoReloadToggle} onReload={this._loadData}/>
+          <div className="filter-block">
+            <MultiSelect
+              maxShowValues={3}
+              allowSelectAll
+              isMulti
+              className="multi-select-container"
+              options={columnOptions}
+              closeMenuOnSelect={false}
+              hideSelectedOptions={false}
+              value={selectedColumns}
+              allOption={allColumnsOption}
+              onChange={this._handleColumnChange}
+            /> Columns
+          </div>
+          <AutoReloadPanel
+            isLoading={loading}
+            lastUpdated={lastUpdated}
+            togglePolling={this.autoReload.handleAutoReloadToggle}
+            onReload={this._loadData}
+          />
         </div>
-        <div className='content-div'>
-          <Table dataSource={dataSource} columns={COLUMNS} loading={loading} pagination={paginationConfig} rowKey='hostname'/>
+
+        <div className="content-div">
+          <Table
+            dataSource={dataSource}
+            columns={COLUMNS.filter((column) =>
+              selectedColumns.some((e) => e.label === column.key)
```

Comment:
nit: compare value instead of label
```suggestion
              selectedColumns.some((e) => e.value === column.key)
```

Reply:
fixed